web_api.md

```diff
@@ -0,0 +1,261 @@
+RabbitMQ will automatically re-queue the message after timeout, which is 30
+minutes [by default](https://www.rabbitmq.com/consumers.html#acknowledgement-timeout). This value is, however, can be
+configured.
+
+To avoid processing the same input twice (in case of re-queuing), a Processing Server first checks
+the [`redeliver`](https://www.rabbitmq.com/confirms.html#automatic-requeueing) property to see if this message was
+re-queued. If yes, and the status of this process in the database is not `SUCCESS`, it will process the data described
+in the message again.
+
+When a Processing Broker receives a request, it adds some more data to the content, then push it to an appropriate
+queue. A processing queue always has the same name as its respective processors. For example, `ocrd-olena-binarize`
+processors listen only to the queue named `ocrd-olena-binarize`. Below is an example of how a message looks like. For a
+detailed schema, please check the [message schema](web_api/processing-message.schema.yml).
+
+```yaml
+job_id: uuid
+processor_name: ocrd-cis-ocropy-binarize
+
+path_to_mets: /path/to/mets.xml
+input_file_grps:
+  - OCR-D-DEFAULT
+output_file_grps:
+  - OCR-D-BIN
+page_id: PHYS_001,PHYS_002
+parameters:
+  params_1: 1
+  params_2: 2
+result_queue_name: ocrd-cis-ocropy-binarize-result
+
+created_time: 1668782988590
+```
+
+In the message content, except `job_id`, `processor_name`, and `created_time`, are added by the Processing Broker. The
+rest comes from the body of the `POST /processor/{executable}` request.
+
+Instead of `path_to_mets`, one can also use `workspace_id` to specify a workspace. An ID of a workspace can be obtained
+from the Workspace Server. In case `result_queue_name` property is presented, the result of the processing will be
+pushed to the queue with the provided name. If the queue does not exist yet, it will be created on the fly. This is
+useful when there is another service waiting for the results of processing. That service can simply listen to that queue
+and will be immediately notified when the results are available. The message schema can be
+found [here](web_api/result-message.schema.yml), while an example of a result message looks like this:
+
+```yaml
+job_id: uuid
+status: SUCCESS
+path_to_mets: /path/to/mets.xml
+```
+
+With the returned `job_id`, one can retrieve more information by sending a `GET` request to
+the `/processor/{executable}/{job_id}` endpoint, or to `/processor/{executable}/{job_id}/log` to get all logs of that
+job.
+
+### Database
+
+The database is required to store necessary information such as users requests, jobs statuses, workspaces, etc. We
+recommend to use [MongoDB](https://www.mongodb.com/) since it is used by Processing Servers, but other kinds of storage
+may work as well. However, there must be a MongoDB running in the system because all Processing Servers will read and
+write to it. To connect to MongoDB via a Graphical User
```

Comment:
This seems contradictory. If MongoDB is required for the broker, why and how could you use "other kinds of storage"?

Reply:
I agree. The text was removed.